storage-proofs/src/stacked/proof.rs

```diff
@@ -360,101 +447,59 @@ impl<'a, H: 'static + Hasher> StackedDrg<'a, H> {
                 None => build_tree(&data),
             };
 
-            // encode layers
-            let encodings = encodings_handle.join().expect("failed to encode layers")?;
-            let size = encodings.encoding_at_last_layer().len();
-
-            // encode original data into the last layer
-            info!("encoding data");
-            encodings
-                .encoding_at_last_layer()
-                .read_range(0..size)
-                .into_par_iter()
-                .zip(data.par_chunks_mut(NODE_SIZE))
-                .try_for_each(|(key, data_node_bytes)| -> Result<()> {
-                    let data_node = H::Domain::try_from_bytes(data_node_bytes)?;
-                    let encoded_node = encode::<H::Domain>(key, data_node);
-
-                    // store result in the data
-                    data_node_bytes.copy_from_slice(AsRef::<[u8]>::as_ref(&encoded_node));
-
-                    Ok(())
-                })?;
-
-            // the last layer is now stored in the data slice
-            let r_last = data;
-
-            // construct final replica commitment
-            let tree_r_last_handle = s.spawn(move |_| build_tree(&r_last));
-
-            // construct column commitments
-            info!("constructing column commitments");
-
-            // For now split into 4 chunks to trade space (memory) vs speed reasonably.
-            let chunks = 4;
-
-            let len = nodes_count * NODE_SIZE;
-            let node_part_len = nodes_count / chunks;
-
-            let mut cs = vec![0; len];
-
-            let part_len = node_part_len * NODE_SIZE;
-            let (p1, p2) = cs.split_at_mut(part_len * 2);
-            let (a, b) = p1.split_at_mut(part_len);
-            let (c, d) = p2.split_at_mut(part_len);
-
-            crossbeam::thread::scope(|s| {
-                let a_handle = s.spawn(|_| {
-                    for (x, chunk) in (0..node_part_len).zip(a.chunks_exact_mut(NODE_SIZE)) {
-                        chunk.copy_from_slice(AsRef::<[u8]>::as_ref(&encodings.column_hash(x)));
-                    }
+            let (encodings, column_hashes) = h.join().unwrap().unwrap();
+            let column_hashes = column_hashes.unwrap();
+
+            (encodings, column_hashes, tree_d)
+        })?;
+
+        let (tree_r_last, tree_c, comm_r): (Tree<H>, Tree<H>, H::Domain) =
```

Comment:
It would be slightly better to change the order to `(tree_c, tree_r_last, comm_r)`. I think if we use this ordering consistently, things will be less confusing, and we'll also be less likely to accidentally mix things up in a way the type system won't help us catch.

Reply:
We don't use this ordering anywhere else, so not sure where the ordering you propse is coming from.